satellite/metainfo/metainfo.go

```diff
@@ -645,15 +645,10 @@ func (endpoint *Endpoint) CreateBucket(ctx context.Context, req *pb.BucketCreate
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	// checks if bucket exists before updates it or makes a new entry
+	// checks if bucket exists before updates it with partner id if it exits or makes a new entry
+	partnerID := keyInfo.PartnerID
 	bucket, err := endpoint.metainfo.GetBucket(ctx, req.GetName(), keyInfo.ProjectID)
 	if err == nil {
```

Comment:
This is actually what @egonelbre 's comments came from a few weeks ago. We should be pulling the partnerID from a user-agent string not the request object.

Reply:
got it..

Reply:
[How to get UserAgent on server side](https://github.com/grpc/grpc-go/issues/1100)

Reply:
Let's do this in this PR